storages/backends/gcloud.py

```diff
@@ -168,10 +178,14 @@ def _save(self, name, content):
         if CONTENT_TYPE not in blob_params:
             upload_params[CONTENT_TYPE] = file_object.mime_type
 
+        if self.gzip and upload_params.get(CONTENT_TYPE) in self.gzip_content_types:
```

Comment:
I'm still not sure about this check.

If the user chooses to put `content_type` in `get_object_parameters`, then `upload_params[CONTENT_TYPE]` would never be set so this wouldn't be hit.

Maybe what we actually need above is

```python
upload_params[CONTENT_TYPE] = blob_params.pop(CONTENT_TYPE, file_object.mime_type)
```

Then I think to mirror S3 what we want is to check

```python
if (self.gzip and 'content_encoding' not in blob_params
       and upload_params[CONTENT_TYPE] in self.gzip_content_types):
  ....
```

What do you think?

Reply:
Fixed. Thanks!